src/charm.py

```diff
@@ -509,15 +492,38 @@ def _on_config_changed(self, event: EventBase):
         self._update_fiveg_n3_relation_data()
         self._update_fiveg_n4_relation_data()
 
+    def _cpu_check(self):
```

Comment:
I would suggest to give a name which is more descriptive such as `_is_cpu_check_passed`.